packages/kong-plugin/kong/plugins/readme-plugin/handler.lua

```diff
@@ -0,0 +1,247 @@
+local Queue = require "kong.tools.queue"
+local cjson = require "cjson"
+local url = require "socket.url"
+local http = require "resty.http"
+
+local kong = kong
+local ngx = ngx
+local encode_base64 = ngx.encode_base64
+local tostring = tostring
+local tonumber = tonumber
+local fmt = string.format
+local pairs = pairs
+local max = math.max
+
+-- this is used so we don't have to parse the url every time we log.
+local parsed_urls_cache = {}
```

Comment:
is there any worry that this grows without limit?

Reply:
Great question! In default operation this should only be size 1. But if they are running many instances of this plugin with different `proxy_endpoint` then this could grow. But I think that is highly unlikly to happen. 

`parse_url` is called on every request to parse our static `https://metrics.readme.io/v1/request"` url (or the `proxy_endpoint` they set as an override to this). 
The readme URL is hardcoded so it won't change, the `proxy_endpoint` can change, but there typically is only a handfull of proxies a company has.